### PR TITLE
'Concatenation of two strings' is added in the 'other' section of C file' #3011

### PR DESCRIPTION
**Pull request by jyoti-anjaly**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes , I have read the guidelines.

### Description

I added a new file in "Other" file in "C" . Concatenation of two strings without using strcat() function.
Issue number: #3011

### Checklist

- [ /] I've read the contribution guidelines.
- [ /] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

This issue was created by me. Issue number #3011

### CODE REVIEW

File: C/other/concatenate_two_strings.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  char str1[100] = "Hey,Open Source is amazing", str2[] = " Start contributing";   //It's important to note that the length of str1 should be sufficient to hold the string after concatenation.
+  int length, j;
+
+  // store length of str1 in the length variable
+  length = 0;
+  while (str1[length] != '\0') {
+    ++length;
+  }
+
+  // concatenate str2 to str1
```
Comment (ankitaggarwal23) [inline C/other/concatenate_two_strings.c]: Remove the redundant comments.

File: C/other/concatenate_two_strings.c [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  for (j = 0; str2[j] != '\0'; ++j, ++length) {
+    str1[length] = str2[j];
+  }
+
+  // terminating the str1 string
```
Comment (ankitaggarwal23) [inline C/other/concatenate_two_strings.c]: Remove the redundant comments.

File: C/other/concatenate_two_strings.c [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  str1[length] = '\0';
+
+  printf("After concatenation, Final string will be: ");
```
Comment (ankitaggarwal23) [inline C/other/concatenate_two_strings.c]: Use `\n` to print the output in new line which looks more readable and understandable.

File: C/other/concatenate_two_strings.c [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  puts(str1);
+
+  return 0;
+}
+//The two strings str1 and str2 will be concatenated and the result will be  stored in s1.
```
Comment (ankitaggarwal23) [inline C/other/concatenate_two_strings.c]: Remove the redundant comments.

File: C/other/concatenate_two_strings.c [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// The best way to concatenate two strings in C programming is by using the strcat() function.But,in this example, we will concatenate two strings manually.
+#include <stdio.h>
+#include <string.h>     //header file
+
+int main() {
+  char str1[100] = "Hey,Open Source is amazing", str2[] = " Start contributing";   //It's important to note that the length of str1 should be sufficient to hold the string after concatenation.
+  int length, j;
+
+  // store length of str1 in the length variable
```
Comment (ankitaggarwal23) [inline C/other/concatenate_two_strings.c]: Remove the redundant comments.

File: C/other/concatenate_two_strings.c [hunk 6]
```diff
@@ -0,0 +1,30 @@
+// The best way to concatenate two strings in C programming is by using the strcat() function.But,in this example, we will concatenate two strings manually.
+#include <stdio.h>
+#include <string.h>     //header file
+
+int main() {
+  char str1[100] = "Hey,Open Source is amazing", str2[] = " Start contributing";   //It's important to note that the length of str1 should be sufficient to hold the string after concatenation.
```
Comment (ankitaggarwal23) [inline C/other/concatenate_two_strings.c]: - Remove the redundant comments.
- Don't fix input from yourself. Read I/P from user for both the strings.

File: C/other/concatenate_two_strings.c [hunk 7]
```diff
@@ -0,0 +1,30 @@
+// The best way to concatenate two strings in C programming is by using the strcat() function.But,in this example, we will concatenate two strings manually.
+#include <stdio.h>
+#include <string.h>     //header file
```
Comment (ankitaggarwal23) [inline C/other/concatenate_two_strings.c]: Remove the redundant comments.

File: C/other/concatenate_two_strings.c [hunk 8]
```diff
@@ -0,0 +1,30 @@
+// The best way to concatenate two strings in C programming is by using the strcat() function.But,in this example, we will concatenate two strings manually.
```
Comment (ankitaggarwal23) [inline C/other/concatenate_two_strings.c:1]: Few changes: 

- Add a beginner-friendly description at the top and make it concise and looks readable.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by ankitaggarwal23** [comment]
@jyoti-anjaly Mark the checklist properly and update the README.md file as well.

Capture

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Kindly Do the changes as mentioned earlier

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Also do the changes marked above
/*
SAMPLE OUTPUT: 

After concatenation, Final string will be:
Hey,Open Source is amazing Start contributing
*/
- Add a  new line after output.

**Comment by jyoti-anjaly** [comment]
@ankitaggarwal23 Sir,I have made changes as told by use. Can you please check this? #3347

**Pull request closed without merging**